service/controller/resource/cloudconfigblob/delete.go

```diff
@@ -0,0 +1,9 @@
+package cloudconfigblob
+
+import (
+	"context"
+)
+
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
```

Comment:
Should we remove the blob when node pool is removed?

Reply:
Yep, I think that would make sense.

Reply:
ok, added the deletion logic. I chose the option to include all the snapshots in the deletion, according to https://docs.microsoft.com/en-us/rest/api/storageservices/delete-blob#request-headers